src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionBase.cs

```diff
@@ -1075,6 +1075,20 @@ protected virtual void DisposeInternal()
                     // ignored
                 }
 
+                try
+                {
+                    if (SubscriptionConnectionTask is { IsCompleted: false })
+                    {
+                        // precaution - it's supposed this task will fail here since we disposed all resources used by connection
+                        // but we should wait for it before we release _copiedBuffer
+                        SubscriptionConnectionTask.Wait();
```

Comment:
isn't this the task that actually called the connection dispose ?

Reply:
Let's talk about that directly

Reply:
Indeed there is a possibility to have a deadlock here. We'll discuss it later.

Reply:
Shouldn't we just always call `Wait()` here?